### DISCUSSION TITLE
Idea for adjusting HantushWellModel for wells in multi-aquifer systems

### DISCUSSION THREAD

**Discussion by ursusmutus** [pastas/pastas#783] [Ideas]
**Describe the proposed enhancement**
adding an optional parameter c in the HantushWellModel rfunc so that
gain = A*K0 * (2(r+c) sqrt(b)

**Expected behavior**
With c we add distance to all wells in the well model. This helps to model multi aquifer systems (e.g. if there is an aquitard between the extractions an the observation well.) 

**Additional context**
Parameter c works as a kind of aquitard resistance so that extractions further away are getting a bit more important over the extractions of the closest wells

For now I'm doing it manually by adding 10, 100, 1000 or even 10000 m to all well distances before modeling and using the model with the best fit with good results.

**Comment by dbrakenhoff** [comment]
Interesting idea @ursusmutus, I'm curious if you're able to get good results using that approach!

In the past I've split wells in different aquifers over different HantushWellModel response functions. So all wells in each aquifer have the same response scaled with distance, but different aquifers can respond differently. From a hydrological point of view, and the current capabilities in Pastas I would think that separate response functions for each aquifer would be the more appropriate choice, but I'm curious to see how your approach performs. Would you be willing to share a comparison between a model using your approach and an approach where each aquifer is modeled separately?

For grouping wells in multi-aquifer systems using a single response function, it would make more sense to use a response function that accounts for that explicitly, but that is not currently implemented in Pastas. We have recently been talking about implementing a Pastas-plugins structure that would allow more custom StressModels to be applied. An example would be an multi-layer analytic element model using [TTim](https://ttim.readthedocs.io/en/latest/). The analytic model is then used to compute a response to pumping. This is a kind of hybrid form of time series modeling and more traditional groundwater modeling. This still requires quite some testing and development however, so this won't be available for some time. 

I'm converting this issue into a discussion for now, as I think this requires a bit more thinking and testing before we have a concrete idea of what we want to implement in Pastas.
